pkg/hubble/peer/service.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package peer
+
+import (
+	peerpb "github.com/cilium/cilium/api/v1/peer"
+	"github.com/cilium/cilium/pkg/node/manager"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// Service implements the peerpb.PeerServer gRPC service.
+// TODO: implement tests. This requires turning manager.Manager into an
+// interface so that it can be mocked for the purpose of testing.
+type Service struct {
+	stop chan (struct{})
+	mgr  *manager.Manager
```

Comment:
Nit, consider using an interface instead of implementation.

Reply:
Yup, this is required for testing, like mentioned in the struct comment just above. If it's fine with you, I'll add an interface in `pkg/node/manager/manager.go`.

Reply:
The interface could be defined here, not in `pkg/node/manager/manager.go`, this allows you to not need to import `pkg/node/manager` in this package.

Reply:
I've added the interface with this commit faae24149afbb9da49688945123c09a79716f903 already and implemented tests of the service.
I think that importing the interface from `pkg/node/manager` is not so bad because it's otherwise really hard to understand who implements it if you're not familiar enough with cilium's codebase.

Reply:
@Rolinh there are tools built for that purpose [0] so there's no point in keeping the interfaces in the same place as 1 of the implementations. What happens if in the future we will have a different node manager that implements the same interface? Where will the interface live?

[0] https://docs.google.com/document/d/1_Y9xCEMj5S-7rv2ooHpZNH15JgRT5iM742gJkw5LtmQ/edit#heading=h.f6hyappclfor

Reply:
I don't mind moving the interface to `pkg/hubble/peer` if you have strong feelings about it; I don't think it matters much.

Reply:
> The interface could be defined here, not in pkg/node/manager/manager.go, this allows you to not need to import pkg/node/manager in this package.

BTW, by moving the interface to `pkg/hubble/peer` I could save on importing `pkg/node/manager` but then I'll have to import `pkg/datapath` as it's required for the interface definition. I'm not "saving" on import anyway.